tensorflow/inference/docker/build_artifacts/sagemaker/python_service.py

```diff
@@ -244,25 +312,28 @@ def _import_custom_modules(self, model_name):
         else:
             log.info("Model-specific inference script for the model {} not found at path {}.".format(model_name, inference_script_path))
 
-    def _cleanup_config_file(self, config_file):
-        if os.path.exists(config_file):
-            os.remove(config_file)
-
     def _handle_invocation_post(self, req, res, model_name=None):
         if SAGEMAKER_MULTI_MODEL_ENABLED:
             if model_name:
-                if model_name not in self._model_tfs_rest_port:
+                if model_name not in self._mme_tfs_instances_status or \
+                        not self._check_pid(self._mme_tfs_instances_status[model_name][0].pid):
+                    with lock():
```

Comment:
Updated PR to use `lock = manager.Lock()` instead.

Reply:
ok, this could technically still be a degredation in MME with instance count 1, and it's unnecessary in that case since there's nothing to sync with.

Could just add an "if instance count is 1" check in that if statement. Alternatively, you could modify "lock" to be a no-op unless in MME insance_count > 1 case.